src/main/java/uk/co/onsdigital/discovery/metadata/api/dao/MetadataDaoImpl.java

```diff
@@ -28,14 +26,35 @@ public long countDataSets() {
     }
 
     @Override
-    public List<DimensionalDataSet> findDataSetsPage(int pageNumber, int pageSize) {
+    public long countDataResources() {
+        return entityManager.createNamedQuery("DataResource.count", Long.class).getSingleResult();
+    }
+
+    @Override
+    public List<DimensionalDataSet> findLegacyDataSetsPage(int pageNumber, int pageSize) {
         final int firstPageOffset = (pageNumber - 1) * pageSize;
         return entityManager.createNamedQuery("DimensionalDataSet.findAll", DimensionalDataSet.class)
```

Comment:
I can change the model and change these as part of this review. Otherwise, we can push this back.